dietpi/dietpi-software

```diff
@@ -3435,6 +3441,87 @@ _EOF_
 
 		fi
 
+		software_id=99 # Prometheus Node Exporter
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+
+			Banner_Installing
+
+			# Use the default port
+			local port="9100"
+
+			local arch=''
+			case $G_HW_ARCH_NAME in
+				'armv6l') arch='armv6';;
+				'armv7l') arch='armv7';;
+				'aarch64') arch='arm64';;
+				'x86_64') arch='amd64';;
+			esac
+
+			# Reinstall: Remove previous instances
+			[[ -e '/usr/local/bin/node_exporter' ]] && G_EXEC rm -R /usr/local/bin/node_exporter
+			[[ -e '/usr/local/bin/raspberrypi_exporter' ]] && G_EXEC rm -R /usr/local/bin/raspberrypi_exporter
+
+			# Download binaries
+			Download_Install $(curl -sSfL 'https://api.github.com/repos/prometheus/node_exporter/releases/latest' | mawk -F\" "/\"browser_download_url\": \".*node_exporter-.*.linux-$arch.tar.gz\"/{print \$4}")
+			G_EXEC mv node_exporter*/node_exporter /usr/local/bin/
+			G_EXEC chmod +x /usr/local/bin/node_exporter
+
+			# User
+			Create_User node_exporter
+
+			# Working directory
+			G_EXEC mkdir -p /var/lib/node_exporter/textfile_collector
```

Comment:
Shall we use `/mnt/dietpi_userdata/node_exporter` as data directory, to align with other software options and use external drive in many cases?

We could also use `/opt/node_exporter` for the binaries then, in case it is not usually intended to be called from console by times.

Reply:
Still suggesting to also move `/var/lib/node_exporter/textfile_collector` to `/mnt/dietpi_userdata/node_exporter`, so we are consistent with other software options 🙂.

Reply:
That folder isn't really userdata, though, it's more like a temporary file. A single file is created in that directory by `raspberrypi_exporter` and will be overwritten every 15 minutes. The contents include cpu frequency stats, temperatures, voltages, etc. That file will be served by `node_exporter` along with the other metrics it gathers.

I can move it to userdata if you really want, but I don't think it really belongs there, the file is useless outside of the particular point in time it's generated.

Reply:
Would it then make sense to store it in `/run/node_exporter` instead and add `RuntimeDirectory=node_exporter` to `node_exporter.service`? So we don't do disk I/O for a temporary file.

The question is only how to configure `raspberrypi_exporter` to write to this directory instead, and also we need to document it, if `/var/lib/node_exporter` is the default and ... okay no chance: https://github.com/fahlke/raspberrypi_exporter#prerequisites
> node_exporter `--collector.textfile.directory` must be set to `/var/lib/node_exporter/textfile_collector`

Only thing we could do is using a symlink.

Reply:
Made a related feature request: https://github.com/fahlke/raspberrypi_exporter/issues/4

Reply:
Ah, it's a bash script with `METRICS_FILE` which can be easily adjusted.